js/angular/services/foundation.dynamicRouting.animations.js

```diff
@@ -1,122 +1,87 @@
+          angular.forEach(cleanup, function (cb) {
+            if (angular.isFunction(cb)) {
+              cb();
+            }
+          });
+
+          destroyed();
+        });
 
-        //start animation
-        element.addClass(animationIn);
-        element.addClass(initial);
+        function onStateChangeStart() {
+          if ($state.includes(getState()) && animation.leave) {
+            element.addClass(animation.leave);
 
-        $rootScope.$digest();
+            var parentHeight = parseInt(element.parent()[0].style.height);
+            var elHeight = parseInt(window.getComputedStyle(element[0], null).getPropertyValue('height'));
 
-        element[0].style.transitionDuration = '';
-        element.addClass(activate);
+            var tempHeight = parentHeight > 0 ? parentHeight : elHeight > 0 ? elHeight : '';
 
-        var finishAnimation = function() {
-          element.parent().removeClass(parentStyle);
-          element.removeClass(activate + ' ' + initial + ' ' + animationIn + ' ' + animationOut);
-          timedOut = false;
-          done();
+            element.parent()[0].style.height = tempHeight + 'px';
+            element.parent().addClass('position-absolute');
+          }
         }
 
-        element.one(events.join(' '), function() {
-          finishAnimation();
-        });
-
-        setTimeout(function() {
-          if (timedOut) {
-            finishAnimation();
+        function onStateChangeError() {
+          if (animation.leave) {
+            element.removeClass(animation.leave);
           }
-        }, 3000);
-
-      } else {
-        done();
-      }
 
-      return function(isCancelled) {
-
-      };
-    }
-
-    function leaveAnimation(element, done) {
-      var scope = element.scope();
-
-      if(scope.vars && scope.vars.animationOut) {
-        var animationIn  = scope.vars.animationIn || '';
-        var animationOut = scope.vars.animationOut;
-        var initial  = 'ng-leave';
-        var activate = 'ng-leave-active';
-        var timedOut = true;
-
-        element.removeClass(activate + ' ' + initial + ' ' + animationIn + ' ' + animationOut);
-        element[0].style.transitionDuration = 0;
+          element.parent().removeClass('position-absolute');
+          element.parent()[0].style.height = null;
+        }
 
-        //start animation
-        element.addClass(animationOut);
-        element.addClass(initial);
+        function onStateChangeSuccess() {
+          if ($state.includes(getState()) && animation.enter) {
+            element.addClass(animation.enter);
+          }
 
-        $rootScope.$digest();
+        }
 
-        element[0].style.transitionDuration = '';
-        element.addClass(activate);
+        function onViewContentAnimationEnded(ev) {
+          if (ev.targetScope === scope && animation.enter) {
+            element.removeClass(animation.enter);
+            element.parent().removeClass('position-absolute');
```

Comment:
@stryju hey so I did some work on the animations and added some important parts (like the `position-absolute` classes). Basically, the idea behind these is this:

1. every ui-view that's changing out has to be absolutely positioned. We can easily achieve this via `element.addClass()` or whatever directly on the element instead of the parent however:
2. ui-view's parent needs to be relatively positioned and have a set height. This is so that the animation can take place without any elements "jumping up" to take the place of the ui-view.

The problem I'm facing here is that `onViewContentAnimationEnded` runs for *both* views (the leaving and entering one). I need to, somehow, make sure that the class removal and height reset happens only *after* the animations take place on both sides. Whatever I tried either left the animation information in (the height and class) or it would be removed too quickly so that an animation would be cut short.

Any ideas?